phpBB/phpbb/user.php

```diff
@@ -281,9 +281,48 @@ function setup($lang_set = false, $style_id = false)
 			$db->sql_freeresult($result);
 		}
 
+		/**
+		 * Something went very bad this time.
+		 * Fallback to board's default style upon its strict verification.
+		 */
+		if (!$this->style)
+		{
+			/** Verify default style exists in the database */
+			$sql = 'SELECT style_id
+				FROM ' . STYLES_TABLE . '
+				WHERE style_id = ' . (int) $config['default_style'];
+			$result = $db->sql_query($sql);
+			$style_id = (int) $db->sql_fetchfield('style_id');
+			$db->sql_freeresult($result);
+
+			$style_id = $style_id ?: false;
```

Comment:
What's the purpose of this line? First of, you shouldn't mix int and bool here. If style id is 0, just keep it at that. It will properly evaluate in the other checks.